components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ServiceCatalog.java

```diff
@@ -123,6 +124,16 @@ ServiceEntry getServiceByUUID(String serviceCatalogId, int tenantId)
      */
     ServiceEntry getServiceByKey(String key, int tenantId) throws APIManagementException;
 
+    /**
+     * Get shrinked information of a service endpoint by key
+     *
+     * @param key Service key unique to each tenant
+     * @param tenantId Tenant Identifier
+     * @return ServiceEntry object including endpoint information
+     * @throws APIManagementException if failed to add ServiceCatalogInfo
+     */
+    ServiceEntryResponse getServiceBasicInfoByKey(String key, int tenantId) throws APIManagementException;
```

Comment:
```suggestion
    ServiceEntryInfo getServiceBasicInfoByKey(String key, int tenantId) throws APIManagementException;
```

Reply:
Changed in f7fb45ff4105968058bbe5639100866b4f012e6e